enterprise/api/v1/views.py

```diff
@@ -510,3 +515,94 @@ class EnterpriseCustomerReportingConfigurationViewSet(EnterpriseReadOnlyModelVie
+        """
+        Gets ``email``, ``enterprise_name``, and ``number_of_codes``,
+        which are the relevant parameters for this API endpoint.
+
+        :param request: The request to this endpoint.
+        :return: The ``email``, ``enterprise_name``, and ``number_of_codes`` from the request.
+        """
+        email = get_request_value(request, self.REQUIRED_PARAM_EMAIL, '')
+        enterprise_name = get_request_value(request, self.REQUIRED_PARAM_ENTERPRISE_NAME, '')
+        number_of_codes = get_request_value(request, self.OPTIONAL_PARAM_NUMBER_OF_CODES, '')
+        if not (email and enterprise_name):
+            raise CodesAPIRequestError(
+                self.get_missing_params_message([
+                    ("'email'", bool(email)),
+                    ("'enterprise_name'", bool(enterprise_name)),
+                ])
+            )
+        return email, enterprise_name, number_of_codes
+
+    def get_missing_params_message(self, parameter_state):
+        """
+        Get a user-friendly message indicating a missing parameter for the API endpoint.
+        """
+        params = ', '.join(name for name, present in parameter_state if not present)
+        return self.MISSING_REQUIRED_PARAMS_MSG.format(params)
+
+    def post(self, request):
+        """
+        POST /enterprise/api/v1/request_codes
+
+        Requires a JSON object of the following format:
+        >>> {
+        >>>     "email": "bob@alice.com",
+        >>>     "enterprise_name": "IBM",
+        >>>     "number_of_codes": "50"
+        >>> }
+
+        Keys:
+        *email*
+            Email of the customer who has requested more codes.
+        *enterprise_name*
+            The name of the enterprise requesting more codes.
+        *number_of_codes*
+            The number of codes requested.
+        """
+        try:
+            email, enterprise_name, number_of_codes = self.get_required_query_params(request)
+        except CodesAPIRequestError as invalid_request:
+            return Response({'error': str(invalid_request)}, status=HTTP_400_BAD_REQUEST)
+
+        subject_line = _('Code Management - Request for more Codes by {token_enterprise_name}').format(
+            token_enterprise_name=enterprise_name
+        )
+        plain_msg = _('{token_email} from {token_enterprise_name} has requested for more {token_number_codes} codes.'
+                      'Please reach out to them.').format(
+                          token_email=email,
+                          token_enterprise_name=enterprise_name,
+                          token_number_codes=number_of_codes
+                      )
+        from_email_address = None
+        cs_email = _('customersuccess@edx.org')
+        data = {
+            self.REQUIRED_PARAM_EMAIL: email,
+            self.REQUIRED_PARAM_ENTERPRISE_NAME: enterprise_name,
+            self.OPTIONAL_PARAM_NUMBER_OF_CODES: number_of_codes,
+        }
+        if mail.send_mail(
```

Comment:
I think we need to wrap this in a try/except, set fail_silently to false, and handle SMTPException so we can capture the email/enterprise name in the error message.  I believe as it is this would swallow the errors.